test/kvao/delete-all.suite.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+const bdd = require('../helpers/bdd-if');
+const helpers = require('./_helpers');
+const should = require('should');
+
+module.exports = function(dataSourceFactory, connectorCapabilities) {
+  var supportsDeleteAllOperation =
+    connectorCapabilities.supportsDeleteAllOperation !== false;
+
+  bdd.describeIf(supportsDeleteAllOperation, 'deleteAll', function() {
+    let CacheItem;
+    beforeEach(function unpackContext() {
+      CacheItem = helpers.givenCacheItem(dataSourceFactory);
+    });
+
+    it('removes all keys-value pairs associated to the given model', function() {
+      return helpers.givenKeys(CacheItem, ['key1', 'key2'])
+        .then(() => CacheItem.deleteAll())
+        .then(() => CacheItem.keys())
+        .then((keys) => {
+          should(keys).eql([]);
+        });
+    });
+
+    it('does not remove keys-value pairs unassociated to the given model',
```

Comment:
`key-value pairs`, not `keys-...`

I also find the difference between test names that say "associated" and "unassociated" too small, it can be easily overlooked. How about

```js
it('does not remove data of other models')
```